tracing/tracers/basic/basic.go

```diff
@@ -21,34 +21,34 @@ func InitTracer(opts []string) (opentracing.Tracer, error) {
 	)
 
 	for _, o := range opts {
-		parts := strings.SplitN(o, "=", 2)
-		switch parts[0] {
+		k, v, _ := strings.Cut(o, "=")
+		switch k {
 		case "drop-all-logs":
 			dropAllLogs = true
 
 		case "sample-modulo":
-			if len(parts) == 1 {
-				return nil, missingArg(parts[0])
+			if v == "" {
```

Comment:
Note: for equivalent refactoring it should check for `found` value as `v` could be empty, e.g. for `strings.Cut("foo=", "=")`.
Here and in most other places it is fine to check for empty `v` as empty value should not be permitted.

Reply:
I think `v == ""` must be kept regardless `found` is `true` or `false`. This particular example `strings.Cut("foo=", "=")` will return `"foo", "", true`.
Also It covers this case `strings.Cut("drop-all-logs", "=")` -> `"drop-all-logs", "", false`, so If we'd introduce a `found` check what should be do in case `found == false` ? Is doing a `continue` enough, so we skip to the next option ?